packages/commerce-sdk-react/src/test-utils.tsx

```diff
@@ -22,27 +22,38 @@ export const TEST_CONFIG = {
     locale: 'en_US',
     currency: 'USD'
 }
-const TestProviders = (props: {children: React.ReactNode}) => {
+const TestProviders = (props: {
+    children: React.ReactNode
+    optionalConfig?: Pick<CommerceApiProviderProps, 'headers' | 'fetchOptions'>
+}) => {
     const queryClient = new QueryClient()
-
+    const {optionalConfig} = props
     return (
         <QueryClientProvider client={queryClient}>
-            <CommerceApiProvider {...TEST_CONFIG}>{props.children}</CommerceApiProvider>
+            <CommerceApiProvider {...TEST_CONFIG} {...optionalConfig}>
+                {props.children}
+            </CommerceApiProvider>
         </QueryClientProvider>
     )
 }
 
 /**
  * Render your component, which will be wrapped with all the necessary Provider components
  *
- * @param component
+ * @param children
  * @param options - additional options for testing-library's render function
+ * @param wrapperProps - additional props to pass to TestProvider component
  */
 export const renderWithProviders = (
```

Comment:
If we follow my above suggestion, then this function would be like:

```suggestion
export const renderWithProviders = (
    children: React.ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>,
    providerProps?: {
        commerceApiProvider?: Partial<CommerceApiProviderProps>
        queryClientProvider?: Partial<QueryClientProviderProps>
    }
): void => {
    render(children, {
        // eslint-disable-next-line react/display-name
        wrapper: () => <TestProviders {...providerProps}>{children}</TestProviders>,
        ...options
    })
}
```